File: cinema-booking-system/backend/services/facility-service/FacilityService.Domain/Entities/SeatType.cs
using System;
using FacilityService.Domain.Enum;

namespace FacilityService.Domain.Entities
{
    public class SeatType
    {
        public long Id { get; private set; }
        public SeatTypeCode Code { get; private set; }
        public string Name { get; private set; }
        public string? DisplayName { get; private set; }
        public decimal? PriceMultiplier { get; private set; }
        public int? DefaultColumnSpan { get; private set; } = 1;
        public string? Description { get; private set; }

        private SeatType() 
        { 
            Name = null!;
        } // EF Core

        public SeatType(SeatTypeCode code, string name, string? displayName, decimal? priceMultiplier, int? defaultColumnSpan, string? description)
        {
            if (string.IsNullOrWhiteSpace(name))
            {
                throw new ArgumentException("Name is required", nameof(name));
            }
            
            Code = code;
            Name = name;
            DisplayName = displayName;
            PriceMultiplier = priceMultiplier;
            DefaultColumnSpan = defaultColumnSpan ?? 1;
            Description = description;
        }

        public void Update(string name, string? displayName, decimal? priceMultiplier, int? defaultColumnSpan, string? description)
        {
            if (string.IsNullOrWhiteSpace(name))
            {
                throw new ArgumentException("Name is required", nameof(name));
            }

            Name = name;
            DisplayName = displayName;
            PriceMultiplier = priceMultiplier;
            DefaultColumnSpan = defaultColumnSpan ?? 1;
            Description = description;
        }
    }
}

File: cinema-booking-system/backend_legacy/src/main/java/com/uit/cinema/facility/entity/SeatType.java
package com.uit.cinema.facility.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "seat_types")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class SeatType {

    public enum SeatTypeCode {
        STANDARD,
        VIP,
        COUPLE
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(unique = true, length = 30)
    private SeatTypeCode code;

    @Column(nullable = false, unique = true, length = 50)
    private String name;

    @Column(length = 100)
    private String displayName;

    @Column(precision = 12, scale = 2)
    private BigDecimal priceMultiplier;

    @Builder.Default
    private Integer defaultColumnSpan = 1;

    @Column(columnDefinition = "TEXT")
    private String description;
}
